### PR TITLE
EZP-25602: Reorganize the subitem components to have an extensible subitem view system

### PR DESCRIPTION
**Pull request by dpobel**
JIRA: https://jira.ez.no/browse/EZP-25602

# Description

This patch makes the subitem list extensible. With it, it's possible to write your own way to display the subitem list. To demonstrate the extensibility, in the screencast I implemented a view that displays random animated GIF (that's completely useless but that was funny ;-)):

[](http://www.youtube.com/watch?v=PWzSHl8Nuvo "Click to play on Youtube.com")

The source for the extension is available in https://github.com/dpobel/PlatformUIExtensionSkeletonBundle/compare/custom_subitemview

# Tests

unit test + manual test

### CODE REVIEW

**Comment by dpobel** [comment]
@yannickroger @StephaneDiot that's not completely finished (there are a few todos left), but still it would be nice to start reviewing that if you have some time.

**Comment by andrerom** [comment]
> @yannickroger @StephaneDiot that's not completely finished (there are a few todos left), but still it would be nice to start reviewing that if you have some time.

Maybe then have todos in the description for easier review ;)
- [ ] add todo's in description

**Comment by dpobel** [comment]
ping @yannickroger @StephaneDiot @andrerom

File: Resources/public/js/views/ez-subitemboxview.js [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+        _uiExpand: function () {
+            this._collapse({
+                'collapseElement': this.get('expandableNode'),
+                'detectElement': this.get('container'),
+                'collapsedClass': 'is-subitembox-collapsed',
+            });
+        },
+
+        /**
+         * tap event handler on switch subitem view link. It sswitches the
```
Comment (yannickroger) [inline Resources/public/js/views/ez-subitemboxview.js]: typo: switches

**Comment by yannickroger** [comment]
+1

**Comment by dpobel** [comment]
ping @andrerom @StephaneDiot if you find some time, please review

**Comment by andrerom** [comment]
+0,5 however from my pov it would be very helpful with changes to docs/ on extensibility topics like this *(on design, and how to use)*

**Comment by StephaneDiot** [comment]
+1

**Pull request merged**